simln-lib/src/sim_node.rs

```diff
@@ -682,24 +682,10 @@ impl SimGraph {
         Ok(SimGraph {
             nodes,
             channels: Arc::new(Mutex::new(channels)),
-            tasks: JoinSet::new(),
+            tasks: TaskTracker::new(),
```

Comment:
To be consistent with `Simulation`, let's pass `tasks` in here as well? 

Even though we'll probably use a different constructor to run with a `SimGraph` when we implement this internally, we probably want other use cases that are using `sim-ln` as a library to have to provide (and manage) their own `tasks` - just gives more flexibility.